tests/run.sh

```diff
@@ -42,25 +42,35 @@ if [ ! -f ~/.ssh/id_rsa.pub ]; then
   ssh-keygen -qb 2048 -t rsa -f ~/.ssh/id_rsa -N "" </dev/zero
 fi
 
-if test -z "${AWS_REGION+x}"
-then
-  echo -e "\\e[36m Calculating the AWS region...\\e[0m"
-  AWS_REGION="$(aws configure get region)" ||
-  AWS_REGION="${AWS_REGION:-us-east-1}"
-fi
-export AWS_DEFAULT_REGION="${AWS_REGION}"
-unset AWS_SESSION_TOKEN
-
-### ASSUME ROLE ###
-echo -e "\\e[36m Setting up AWS credentials...\\e[0m"
-ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
-ROLE_ARN="arn:aws:iam::${ACCOUNT_ID}:role/tf-tectonic-installer"
-RES="$(aws sts assume-role --role-arn="${ROLE_ARN}" --role-session-name="jenkins-${CLUSTER_NAME}" --query Credentials --output json)" &&
-export AWS_SECRET_ACCESS_KEY="$(echo "${RES}" | jq --raw-output '.SecretAccessKey')" &&
-export AWS_ACCESS_KEY_ID="$(echo  "${RES}" | jq --raw-output '.AccessKeyId')" &&
-export AWS_SESSION_TOKEN="$(echo "${RES}" | jq --raw-output '.SessionToken')" &&
-CONFIGURE_AWS_ROLES=True ||
-CONFIGURE_AWS_ROLES=False
+case "${BACKEND}" in
+aws)
+  if test -z "${AWS_REGION+x}"
+  then
+    echo -e "\\e[36m Calculating the AWS region...\\e[0m"
+    AWS_REGION="$(aws configure get region)" ||
+    AWS_REGION="${AWS_REGION:-us-east-1}"
+  fi
+  export AWS_DEFAULT_REGION="${AWS_REGION}"
+  unset AWS_SESSION_TOKEN
+
+  ### ASSUME ROLE ###
+  echo -e "\\e[36m Setting up AWS credentials...\\e[0m"
+  ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
+  ROLE_ARN="arn:aws:iam::${ACCOUNT_ID}:role/tf-tectonic-installer"
+  RES="$(aws sts assume-role --role-arn="${ROLE_ARN}" --role-session-name="jenkins-${CLUSTER_NAME}" --query Credentials --output json)" &&
+  export AWS_SECRET_ACCESS_KEY="$(echo "${RES}" | jq --raw-output '.SecretAccessKey')" &&
+  export AWS_ACCESS_KEY_ID="$(echo  "${RES}" | jq --raw-output '.AccessKeyId')" &&
+  export AWS_SESSION_TOKEN="$(echo "${RES}" | jq --raw-output '.SessionToken')" &&
+  CONFIGURE_AWS_ROLES=True ||
+  CONFIGURE_AWS_ROLES=False
+  ;;
+libvirt)
+  [ -z ${IMAGE_PATH+x} ] && (echo "Please set libvirt IMAGE_PATH" >&2; exit 1)
```

Comment:
> What's the `+x` for?

If the parameter is set, substitute `x`.  Docs [here][1].  This line is like [these master lines][2].

[1]: http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02
[2]: https://github.com/openshift/installer/blob/d316449bbf420926b4d927ec7e29d19828e531f3/tests/run.sh#L13

Reply:
I should have been a bit clearer :-)
Why have you chosen to use this substitution here? It doesn't make sense to me that we would continue even though IMAGE_PATH="" is set.

Reply:
> Why have you chosen to use this substitution here? It doesn't make sense to me that we would continue even though IMAGE_PATH="" is set.

The existing installer had a pattern of this usage, so I followed that pattern.  If the user wants to explicitly set `IMAGE_PATH` empty, I'm fine letting them crash latter when we `tectonic init` (or wherever they'll crash).  It doesn't seem like a high-probability situation.  But if you want the script itself to reject empty-string environment variables, I can update all of these usages to `:+x`.

Reply:
In short the pattern enforces setting any value while explicitly allowing the empty string. Even though the use-case is a bit far fetched it's comprehensible for a testing script. Thanks for being patient here :-)

Reply:
> In short the pattern enforces setting any value while explicitly allowing the empty string.

Yeah.  But this script is not validating the content (e.g. we could check whether a file exists at the configured path, or that that file was readable, or that that file was an image type supported by libvirt).  But we don't do any of that, because in a second or so, some of the validation will happen in the Go installer, and then in a few more seconds some more validation will happen in Terraform and libvirt.

> Even though the use-case is a bit far fetched it's comprehensible for a testing script.

So am I fine as I stand?  Or do you want `:x` -> `:+x` for this and all similar comparisons?  Or something else?

Reply:
I think it's fine as is. We shall wait for the next person asking :-D